src/providers/user-provider.tsx

```diff
@@ -0,0 +1,36 @@
+"use client";
+
+import React, { createContext, useContext } from "react";
+
+interface User {
```

Comment:
write the interfaces in the types dir under relevant context for eg. types/user or types/dashboard...